### ISSUE TITLE
[bug] tree-sitter CSS cannot parse pseudo-class with strings

### ISSUE THREAD

**Issue by mohamedmansour** [ast-grep/ast-grep#1569]
### Please read the FAQ for the bug you encountered.

- [X] I have read the existing FAQ

### ⏯ Playground Link

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6ImNzcyIsInF1ZXJ5IjoiaW1wb3J0IHByb2Nlc3MgZnJvbSAnbm9kZTpwcm9jZXNzJyIsInJld3JpdGUiOiIiLCJzdHJpY3RuZXNzIjoic21hcnQiLCJzZWxlY3RvciI6IiIsImNvbmZpZyI6IiIsInNvdXJjZSI6IiAgICA6aG9zdChbZm9vPSdiYXInXTpob3Zlcikge1xuICAgICAgYmFja2dyb3VuZC1jb2xvcjogaG90cGluaztcbiAgICB9In0=

### 💻 Code

```
    :host([foo='bar']:hover) {
      background-color: hotpink;
    }
```

### 🙁 Actual behavior

Error Nodes,  we cannot have `pseudo_class_selector` 

### 🙂 Expected behavior

`pseudo_class_selector` arguments should be allowed to have sub `pseudo_class_selectors`. This fails in both playground and node api.

In addition, It should accept quotes , both single and double. Currently if I add `single quotes` in nodejs for a simple `pseudo_class_selector` it fails, meanwhile in playground it doesn't.

### Additional information about the issue

It only happens after an `attribute_selector` has a `class_name` within a `pseudo_class_selector` arguments.

**Comment by mohamedmansour** [comment]
Filed the single quotes failure upstream in tree-sitter since it can be reproduced there: tree-sitter/tree-sitter-css#56

**Comment by mohamedmansour** [comment]
Filled the attribute selector within a pseudo selector failing to parse to upstream in tree-sitter since it can be reproduced there as well tree-sitter/tree-sitter-css#57

**Comment by mohamedmansour** [comment]
@HerringtonDarkholme tree-sitter-css published version 0.23.1, which fixes a lot of bugs in css grammar, so when that merges here, it will fix this bug
